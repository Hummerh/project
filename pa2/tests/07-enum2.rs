enum Colors {
    Red,
    Blue,
    Green
}

enum Numbers {
    One,
    Two,
    Three
}

enum ABCs { A, B, C }

fn foo() -> Colors { Colors::Red }
fn bar(c : Colors) -> Numbers { Numbers::One }

fn main() {
      let x : Colors = Colors::Green;
      let y : Numbers = Numbers::Two;
      let z : ABCs = ABCs::A;
}

