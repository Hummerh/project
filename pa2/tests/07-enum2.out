(crate
    (items
        (enum-def
            (id
                (Colors))
            (enum-ctor-defs
                (enum-ctor-def
                    (id
                        (Red)))
                (enum-ctor-def
                    (id
                        (Blue)))
                (enum-ctor-def
                    (id
                        (Green)))))
        (enum-def
            (id
                (Numbers))
            (enum-ctor-defs
                (enum-ctor-def
                    (id
                        (One)))
                (enum-ctor-def
                    (id
                        (Two)))
                (enum-ctor-def
                    (id
                        (Three)))))
        (enum-def
            (id
                (ABCs))
            (enum-ctor-defs
                (enum-ctor-def
                    (id
                        (A)))
                (enum-ctor-def
                    (id
                        (B)))
                (enum-ctor-def
                    (id
                        (C)))))
        (fn-def
            (id
                (foo))
            (id
                (Colors))
            (block
                (enum
                    (enum-ctor
                        (id
                            (Colors))
                        (id
                            (Red))))))
        (fn-def
            (id
                (bar))
            (fn-params
                (fn-param
                    (pat-id
                        (id
                            (c)))
                    (id
                        (Colors))))
            (id
                (Numbers))
            (block
                (enum
                    (enum-ctor
                        (id
                            (Numbers))
                        (id
                            (One))))))
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (x)))
                    (id
                        (Colors))
                    (enum
                        (enum-ctor
                            (id
                                (Colors))
                            (id
                                (Green)))))
                (let
                    (pat-id
                        (id
                            (y)))
                    (id
                        (Numbers))
                    (enum
                        (enum-ctor
                            (id
                                (Numbers))
                            (id
                                (Two)))))
                (let
                    (pat-id
                        (id
                            (z)))
                    (id
                        (ABCs))
                    (enum
                        (enum-ctor
                            (id
                                (ABCs))
                            (id
                                (A)))))))))